packages/cli/tests/e2e/multienv/TestRemoteHappyPath.tests.ts

```diff
@@ -0,0 +1,250 @@
+
+      // add env
+      result = await execAsync(`teamsfx env add ${env} --env dev`, {
+        cwd: projectPath,
+        env: processEnv,
+        timeout: 0,
+      });
+      console.log(`[Successfully] env add, stdout: '${result.stdout}', stderr: '${result.stderr}'`);
+
+      // update SKU from free to B1 to prevent free SKU limit error
+      await setSimpleAuthSkuNameToB1Bicep(projectPath, env);
+      console.log(`[Successfully] update simple auth sku to B1`);
+
+      // set active env
+      result = await execAsync(`teamsfx env activate ${env}`, {
+        cwd: projectPath,
+        env: processEnv,
+        timeout: 0,
+      });
+      console.log(
+        `[Successfully] env activate, stdout: '${result.stdout}', stderr: '${result.stderr}'`
+      );
+
+      // list env
+      result = await execAsync(`teamsfx env list`, {
+        cwd: projectPath,
+        env: processEnv,
+        timeout: 0,
+      });
+      const envs = result.stdout.trim().split(/\r?\n/).sort();
+      expect(envs).to.deep.equal(["dev", "e2e (active)"]);
+      expect(result.stderr).to.be.empty;
+      console.log(
+        `[Successfully] env list, stdout: '${result.stdout}', stderr: '${result.stderr}'`
+      );
+
+      // show env
+      result = await execAsync(`teamsfx env`, {
+        cwd: projectPath,
+        env: processEnv,
+        timeout: 0,
+      });
+      expect(result.stdout).to.equal("e2e\n");
+      expect(result.stderr).to.be.empty;
+      console.log(
+        `[Successfully] env show, stdout: '${result.stdout}', stderr: '${result.stderr}'`
+      );
+
+      // provision
+      result = await execAsyncWithRetry(`teamsfx provision`, {
+        cwd: projectPath,
+        env: processEnv,
+        timeout: 0,
+      });
+      console.log(
+        `[Successfully] provision, stdout: '${result.stdout}', stderr: '${result.stderr}'`
+      );
+
+      {
+        // Validate provision
+        // Get context
+        const contextResult = await loadContext(projectPath, env);
+        if (contextResult.isErr()) {
+          throw contextResult.error;
+        }
+        const context = contextResult.value;
+
+        // Validate Aad App
+        const aad = AadValidator.init(context, false, AppStudioLogin);
+        await AadValidator.validate(aad);
+
+        // Validate Simple Auth
+        const simpleAuth = SimpleAuthValidator.init(context);
+        await SimpleAuthValidator.validate(simpleAuth, aad, "B1", true);
+
+        // Validate Tab Frontend
+        const frontend = FrontendValidator.init(context);
+        await FrontendValidator.validateProvision(frontend);
+
+        // Validate Function App
+        const func = FunctionValidator.init(context);
+        await FunctionValidator.validateProvision(func, false);
+
+        // Validate Bot Provision
+        const bot = BotValidator.init(context);
+        await BotValidator.validateProvision(bot);
+      }
+
+      // deploy
```

Comment:
will the e2e test cases cover `publish` operation?

Reply:
Yes, it will be added in next few PRs.